src/backend/kms/surface/timings.rs

```diff
@@ -207,15 +207,21 @@ impl Timings {
             return None;
         }
 
-        Some(
-            self.previous_frames
-                .iter()
-                .rev()
-                .take(window)
-                .map(|f| f.frame_time())
-                .sum::<Duration>()
-                / (window.min(self.previous_frames.len()) as u32),
-        )
+        Some({
+            let count = window.min(self.previous_frames.len()) as u32;
+            let mut total = Duration::ZERO;
+
+            for frame in self.previous_frames.iter().rev().take(window) {
+                if let Some(new_total) = total.checked_add(frame.frame_time()) {
```

Comment:
Nitpicking, but given we can use an iterator chain here, I would kinda prefer this to use `try_fold` to aggregate and return early for `None`.

So something like:
```rust
        Some(
            self.previous_frames
                .iter()
                .rev()
                .take(window)
                .map(|f| f.frame_time())
                .try_fold(Duration::ZERO, |acc, &x| acc.checked_add(x))?
                / (window.min(self.previous_frames.len()) as u32),
        )
```

Reply:
For what it is worth, I still think we should catch this using `try_fold` and also adding a warning like you did in https://github.com/pop-os/cosmic-comp/issues/1062#issuecomment-2735450019 for debug builds at least.

Feel free to update this PR to do that, if you feel like it. Otherwise I'll get to it eventually, but a smithay update should fix the underlying issue once https://github.com/Smithay/smithay/pull/1685 is merged.